node/pkg/transfer-verifier/transfer-verifier-sui-structs.go

```diff
@@ -0,0 +1,436 @@
+	TryMultiGetPastObjects(objectId string, version string, previousVersion string) (SuiTryMultiGetPastObjectsResponse, error)
+}
+
+// This struct defines the standard properties that get returned from the RPC.
+// It includes the ErrorMessage and Error fields as well, with a standard implementation
+// of a `GetError()` function. `suiApiRequest` requires `GetError()` for standard
+// API error handling.
+type SuiApiStandardResponse struct {
+	Jsonrpc string `json:"jsonrpc"`
+	ID      int    `json:"id"`
+	// error_msg is typically populated when a non-api-related error occurs (like ratelimiting)
+	ErrorMessage *string `json:"error_msg"`
+	// error is typically populated when an api-related error occurs
+	Error *struct {
+		Code    int    `json:"code"`
+		Message string `json:"message"`
+	} `json:"error"`
+}
+
+func (e SuiApiStandardResponse) GetError() error {
+	if e.ErrorMessage != nil {
+		return fmt.Errorf("error from Sui RPC: %s", *e.ErrorMessage)
+	}
+
+	if e.Error != nil {
+		return fmt.Errorf("error from Sui RPC: %s", e.Error.Message)
+	}
+
+	return nil
+}
+
+// The response object for suix_queryEvents
+type SuiQueryEventsResponse struct {
+	SuiApiStandardResponse
+	Result SuiQueryEventsResult `json:"result"`
+}
+
+type SuiQueryEventsResult struct {
+	Data []SuiEvent `json:"data"`
+}
+
+type SuiEvent struct {
+	ID struct {
+		TxDigest *string `json:"txDigest"`
+		EventSeq *string `json:"eventSeq"`
+	} `json:"id"`
+	PackageID         *string `json:"packageId"`
+	TransactionModule *string `json:"transactionModule"`
+	Sender            *string `json:"sender"`
+	Type              *string `json:"type"`
+	// Bcs               *string          `json:"bcs"`
+	Timestamp *string          `json:"timestampMs"`
+	Message   *WormholeMessage `json:"parsedJson"`
+}
+
+// The response object for sui_GetTransactionBlock
+type SuiGetTransactionBlockResponse struct {
+	SuiApiStandardResponse
+	Result SuiGetTransactionBlockResult `json:"result"`
+}
+
+type SuiGetTransactionBlockResult struct {
+	ObjectChanges []ObjectChange `json:"objectChanges"`
+	Events        []SuiEvent     `json:"events"`
+}
+
+type ObjectChange struct {
+	ObjectType      string `json:"objectType"`
+	ObjectId        string `json:"objectId"`
+	Version         string `json:"version"`
+	PreviousVersion string `json:"previousVersion"`
+}
+
+// Validate the type information of the object change. The following checks are performed:
+//   - pass the object through a regex that extractts the package ID, coin type, and asset type
+//   - ensure that the asset type is wrapped or native
+//   - ensure that the package IDs match the expected package ID
+//   - ensure that the coin types match
+func (o ObjectChange) ValidateTypeInformation(expectedPackageId string) (success bool) {
+	// AI generated regex
+	re := regexp.MustCompile(`^0x2::dynamic_field::Field<([^:]+)::token_registry::Key<([^>]+)>, ([^:]+)::([^<]+)<([^>]+)>>$`)
```

Comment:
I think this panics on error.  That seems a bit extreme?  Should we consider something less extreme?

Reply:
I was thinking about this, and the thing is that the actual regex remains static. Is there a reason it would compile during tests, but not during a live production run?